packages/@glimmer/component/src/tracked.ts

```diff
@@ -78,29 +75,16 @@ class Tracker {
  *    }
  *  }
  * ```
- *
- * @param dependencies Optional dependents to be tracked.
  */
-export function tracked(...dependencies: string[]): MethodDecorator;
 export function tracked(target: any, key: any): any;
 export function tracked(target: any, key: any, descriptor: PropertyDescriptor): PropertyDescriptor;
-export function tracked(...dependencies: any[]): any {
-  let [target, key, descriptor] = dependencies;
-
-  if (typeof target === "string") {
-    return function(target: any, key: string | Symbol, descriptor: PropertyDescriptor) {
-      if (dependencies.length) {
-        console.warn('The use of dependent keys in @tracked is deprecated (e.g. `@tracked(\'myDependendentKey\')`) and will be removed in v0.12.0');
-        console.warn(`Please remove the "${dependencies.length > 1 ? dependencies.join(', ') : dependencies[0]}" key${dependencies.length > 1 ? 's' : ''}`);
-      }
-      return descriptorForTrackedComputedProperty(target, key, descriptor, dependencies);
-    };
+export function tracked(...args: any[]): any {
+  let [target, key, descriptor] = args;
```

Comment:
Can we add an error somewhere in here to make sure that the usage of `@tracked` is correct? For example:

```
class Foo {
  @tracked() bar = 'foo';
  @tracked('fiz') get bam() {
    return this.fiz + 1;
  }
}
```

These 2 usages are going to result in broken behavior and we should be able to detect this.